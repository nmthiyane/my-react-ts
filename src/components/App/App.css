h1, h2 {
  margin: 5px;
}

.App {
  text-align: center;
  display: block;
  background-color: #9bbcd3;
  height: 1310px;
}

.App-logo {
  float: left;
  animation: App-logo-spin infinite 0.2s linear;
  height: 80px;
}

.App-header {
  background-color: #014373;
  height: 80px;
  padding: 20px;
  color: white;
}

.Components{
  text-align: center;
  display: block;
  min-width: 1500px;
  padding: 0px 1% 0px 1%;
}

.App-title {
  margin-top: 10px;
  font-size: 1.5em;
}

.App-sub-title {
  font-size: 1.2em;
  font-weight: lighter;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(-10deg); }
  to { transform: rotate(5deg); }
}