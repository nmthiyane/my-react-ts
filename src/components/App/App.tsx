import * as React from 'react';
import './App.css';

import { Race, Header, Controls } from '../';

interface State {
	raceInProgress: boolean;
	raceStartPosition: boolean;
	lastWinner: string;
}

export class App extends React.Component<{}, State> {
	constructor(props: {}) {
		super(props);

		this.state = {
			raceInProgress: false,
			raceStartPosition: true,
			lastWinner: ''
		};

		this.setWinner = this.setWinner.bind(this);
		this.startRace = this.startRace.bind(this);
		this.resetRace = this.resetRace.bind(this);
	}

	setWinner(lastWinner: string) {
		this.setState({ lastWinner, raceInProgress: false });
	}

	startRace() {
		this.setState({ ...this.state, raceInProgress: true, raceStartPosition: false });
	}

	resetRace() {
		this.setState({ ...this.state, raceStartPosition: true, lastWinner: '' });
	}

	render() {
		const { lastWinner, raceInProgress, raceStartPosition } = this.state;

		return (
			<div className="App" >
				<Header
					raceInProgress={raceInProgress}
					lastWinner={lastWinner}
				/>
				<article>
					<section>
						<Controls
							resetRace={this.resetRace}
							raceStartPosition={raceStartPosition}
							raceInProgress={raceInProgress}
							startRace={this.startRace}
						/>
						<Race
							raceInProgress={raceInProgress}
							raceStartPosition={raceStartPosition}
							setWinner={this.setWinner}
						/>
					</section>
				</article>
			</div>
		);
	}
}