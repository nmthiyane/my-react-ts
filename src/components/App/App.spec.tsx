import * as React  from 'react';
import { App } from './App';
const renderer  = require('react-test-renderer');

describe('Snapshot test for the Header component', () => {

	it('renders a snapshot', () => {
	const tree = renderer.create((
		<App
		/>
		)).toJSON();
	expect(tree).toMatchSnapshot();
	});
});