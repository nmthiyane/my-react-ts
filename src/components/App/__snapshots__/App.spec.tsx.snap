// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the Header component renders a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src=""
    />
    <h1
      className="App-title"
    >
      Implementation Race
    </h1>
    <div
      className="ScoreBoard"
    >
      <h2
        className="grads"
      >
        Kurtosys Grads
      </h2>
    </div>
  </header>
  <article>
    <section>
      <div
        className="Controls"
      >
        <button
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Start
        </button>
        <button
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Reset
        </button>
      </div>
      <div
        className="Race"
      >
        <div>
          <div
            className="Track"
          >
            <div
              className="Horse"
              style={
                Object {
                  "paddingLeft": "0px",
                }
              }
            >
              <img
                height="100px"
                src="https://avatars3.githubusercontent.com/u/15605543?v=4"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="Track"
          >
            <div
              className="Horse"
              style={
                Object {
                  "paddingLeft": "0px",
                }
              }
            >
              <img
                height="100px"
                src="https://avatars2.githubusercontent.com/u/2463616?v=4"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="Track"
          >
            <div
              className="Horse"
              style={
                Object {
                  "paddingLeft": "0px",
                }
              }
            >
              <img
                height="100px"
                src="https://avatars0.githubusercontent.com/u/1819122?v=4"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="Track"
          >
            <div
              className="Horse"
              style={
                Object {
                  "paddingLeft": "0px",
                }
              }
            >
              <img
                height="100px"
                src="https://avatars2.githubusercontent.com/u/2323810?v=4"
              />
            </div>
          </div>
        </div>
        <div>
          <div
            className="Track"
          >
            <div
              className="Horse"
              style={
                Object {
                  "paddingLeft": "0px",
                }
              }
            >
              <img
                height="100px"
                src="https://avatars3.githubusercontent.com/u/17589003?v=4"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </article>
</div>
`;
